### PR TITLE
PODAUTO-21: Adjust e2e tests so they can run in OpenShift / OpenShif tCI

### PR DESCRIPTION
**Pull request by jkyros**
This is a minimal set of "hacks" that enables the keda e2e suite to run in OpenShift CI. It is not beautiful. 

What it does: 
- adjusts the `Makefile` to allow for image overrides in CI
- allows the tests to run privileged/anyuid pods for now until we can refactor  (centrialized via the helpers) 
- allows replacing images with images we control to avoid pull limits, etc (centralized via the helpers) 
- adds security context to a few of the test pod specs that required them 
- fakes out the prometheus sequential buildinfo test until we can inject the proper git commit into the metrics image from CI 

I'm not in love with it, but it looks like the minimal required to get clean runs without refactoring parts of the test suite.

### CODE REVIEW

**Comment by jkyros** [comment]
So with these changes, the tests run and work, with the exception of the CPU test. 

The CPU test is having trouble getting its metrics for some reason, it looks like:
```
99s         Warning   FailedGetResourceMetric        horizontalpodautoscaler/keda-hpa-cpu-test-so   failed to get cpu utilization: unable to get metrics for resource cpu: no metrics returned from resource metrics API
99s         Warning   FailedComputeMetricsReplicas   horizontalpodautoscaler/keda-hpa-cpu-test-so   invalid metrics (1 invalid out of 1), first error is: failed to get cpu resource metric value: failed to get cpu utilization: unable to get metrics for resource cpu: no metrics returned from resource metrics API
8s          Warning   FailedGetResourceMetric        horizontalpodautoscaler/keda-hpa-cpu-test-so   failed to get cpu utilization: did not receive metrics for targeted pods (pods might be unready)
8s          Warning   FailedComputeMetricsReplicas   horizontalpodautoscaler/keda-hpa-cpu-test-so   invalid metrics (1 invalid out of 1), first error is: failed to get cpu resource metric value: failed to get cpu utilization: did not receive metrics for targeted pods (pods might be unready)
```

Pods are ready, but no metrics. And yeah, the HPA agrees:
```
    - lastTransitionTime: "2023-09-26T17:34:31Z"
      message: 'the HPA was unable to compute the replica count: failed to get cpu
        utilization: unable to get metrics for resource cpu: no metrics returned from
        resource metrics API'
      reason: FailedGetResourceMetric
      status: "False"
      type: ScalingActive
```

File: tests/sequential/prometheus_metrics/prometheus_metrics_test.go [hunk 1]
```diff
@@ -642,6 +643,13 @@ func checkBuildInfo(t *testing.T, families map[string]*prommodel.MetricFamily) {
 	}
 
 	latestCommit := getLatestCommit(t)
+
+	// I haven't found a way to inject the actual value via CI yet, so for now just
+	// tolerate our "we build this in CI" dummy string
+	if _, ok := os.LookupEnv("OPENSHIFT_CI"); ok {
+		latestCommit = "dummy-ci-commit-value"
```
Comment (aravindhp) [inline tests/sequential/prometheus_metrics/prometheus_metrics_test.go:650]: The test Pod in CI will have the repo checked out locally. You can add some code in the release repo to export the latest commit hash and then read it here. I am not asking for it to be done immediately but just something to consider for the future.

**Comment by joelsmith** [comment]
This is a great starting point. Thanks!

/lgtm

File: tests/sequential/prometheus_metrics/prometheus_metrics_test.go (hunk 1 above)
Comment (jkyros) [reply tests/sequential/prometheus_metrics/prometheus_metrics_test.go:650]: I would _love_ to do exactly as you describe, but I think I have the "reverse" of the problem you're describing, i.e. https://github.com/openshift/release/pull/43390#discussion_r1339355574  -- our tests know the info, but our images don't without it being injected ?

**Pull request merged**